src/tests/unit_tls.cpp

```diff
@@ -218,164 +223,216 @@ Test::Result test_tls_handshake(Botan::TLS::Protocol_Version offer_version,
+                        needed = server->received_data(input.data(), input.size());
+                        total_consumed += needed;
+                        }
+                     }
+                  else
+                     {
+                     size_t needed = server->received_data(input.data(), input.size());
+                     result.test_eq("full packet received", needed, 0);
+                     }
+
+                  continue;
+                  }
 
-               if(corrupt_client_data)
+               if(s2c_traffic.size() > 0)
                   {
-                  input = Test::mutate_vec(input, true);
-                  size_t needed = client.received_data(input.data(), input.size());
+                  std::vector<byte> input;
+                  std::swap(s2c_traffic, input);
 
-                  size_t total_consumed = 0;
+                  if(corrupt_client_data)
+                     {
+                     input = Test::mutate_vec(input, true);
+                     size_t needed = client->received_data(input.data(), input.size());
 
-                  while(needed > 0 && result.test_lt("Never requesting more than max protocol len", needed, 18*1024))
+                     size_t total_consumed = 0;
+
+                     while(needed > 0 && result.test_lt("Never requesting more than max protocol len", needed, 18*1024))
+                        {
+                        input.resize(needed);
+                        Test::rng().randomize(input.data(), input.size());
+                        needed = client->received_data(input.data(), input.size());
+                        total_consumed += needed;
+                        }
+                     }
+                  else
                      {
-                     input.resize(needed);
-                     Test::rng().randomize(input.data(), input.size());
-                     needed = client.received_data(input.data(), input.size());
-                     total_consumed += needed;
+                     size_t needed = client->received_data(input.data(), input.size());
+                     result.test_eq("full packet received", needed, 0);
                      }
+
+                  continue;
                   }
-               else
+
+               if(client_recv.size())
                   {
-                  size_t needed = client.received_data(input.data(), input.size());
-                  result.test_eq("full packet received", needed, 0);
+                  result.test_eq("client recv", client_recv, server_sent);
                   }
 
-               continue;
-               }
-
-            if(client_recv.size())
-               {
-               result.test_eq("client recv", client_recv, server_sent);
-               }
-
-            if(server_recv.size())
-               {
-               result.test_eq("server recv", server_recv, client_sent);
-               }
+               if(server_recv.size())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if|for|do|while|switch statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=cpp%3AS134)